### PR TITLE
PR#2123 STS not on class path when using parallel stream

### PR DESCRIPTION
**Pull request by joviegas**
## Description

STS not on class path when using parallel stream

## Motivation and Context

Please check for the context in [PR#2123](https://github.com/aws/aws-sdk-java-v2/issues/2123)
Fixing this issue by changing the Classloader to use ClassLoaderHelper.

Reference material :- https://stackoverflow.com/questions/1771679/difference-between-threads-context-class-loader-and-normal-classloader

## Testing

- Added JUNit
- sudo mvn clean install -pl :sts -P quick --am
- sudo mvn clean install -pl :core -P quick --am

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml [hunk 1]
```diff
@@ -387,6 +387,11 @@
 
         <!-- Checks that no 'final' are added on local variables -->
         <module name="software.amazon.awssdk.buildtools.checkstyle.UnnecessaryFinalOnLocalVariableCheck" />
+
+        <!-- Checks that no forbidden methods are used in implementation -->
+        <module name="com.github.sevntu.checkstyle.checks.coding.ForbidCertainMethodCheck">
```
Comment (zoewangg) [inline build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml:392]: Can we use `Regexp` instead?  That way, we don't need to add another dependency

Example: https://github.com/aws/aws-sdk-java-v2/blob/8f361fde4800451d99dce4a1a22c5557fee61aee/build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml#L361-L365

File: services/sts/src/test/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityCredentialsProviderFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+public class StsWebIdentityCredentialsProviderFactoryTest {
+
+
+    @Test
+    public void stsWebIdentityCredentialsProviderFactory_with_webIdentityCredentialsUtils(){
+        final WebIdentityTokenCredentialsProviderFactory factory = WebIdentityCredentialsUtils.factory();
```
Comment (zoewangg) [inline services/sts/src/test/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityCredentialsProviderFactoryTest.java:31]: nit: we don't use `final` on local variables. We have a checkstyle for it, but it doesn't apply to tests

**Comment by zoewangg** [comment]
Can we add a changelog entry?

File: build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml (hunk 1 above)
Comment (joviegas) [reply build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml]: Thanks for the reference. Changes done.

File: services/sts/src/test/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityCredentialsProviderFactoryTest.java (hunk 2 above)
Comment (joviegas) [reply services/sts/src/test/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityCredentialsProviderFactoryTest.java]: done.

**Comment by joviegas** [comment]
> Can we add a changelog entry?

Done.

**Review by zoewangg — APPROVED** [review]

**Pull request merged**